packages/ember-routing/lib/system/router.js

```diff
@@ -858,8 +858,9 @@ const EmberRouter = EmberObject.extend(Evented, {
       for (let i = 0; i < qpMeta.qps.length; i++) {
         let qp = qpMeta.qps[i];
         let urlKey = qp.urlKey;
+        let qpOther = qpsByUrlKey[urlKey];
 
-        if (qpsByUrlKey[urlKey]) {
+        if (qpOther && (qpOther.hasOwnProperty('controllerName') && qpOther.controllerName !== qp.controllerName)) {
```

Comment:
I'm unsure the `hasOwnProperty` check here is necessary. Ideally if a collision happens, and the property isn't there then we should do the assertion.